src/Services/AuthenticationService.php

```diff
@@ -173,7 +232,31 @@ public function getSessionId(): ?string
     }
 
     /**
-     * Store Relying Party Association to the current session.
+     * @param   ServerRequestInterface  $request
+     * @param   array                   $loginParams
+     *
+     * @return void
+     * @throws Error\BadRequest
+     * @throws Error\NotFound
+     * @throws \JsonException
+     * @throws \SimpleSAML\Module\oidc\Server\Exceptions\OidcServerException
+     */
+
+    public function authenticate(
+        ServerRequestInterface $request,
+        array $loginParams = [],
+    ): void {
+        $oidcClient = $this->getClientFromRequest($request);
+        $authSimple = $this->authSimpleFactory->build($oidcClient);
+
+        $this->sessionService->setIsCookieBasedAuthn(false);
+        $this->sessionService->setIsAuthnPerformedInPreviousRequest(true);
```

Comment:
@cicnavi I'm not familiar with the purpose of `setIsAuthnPerformedInPreviousRequest`, so I'll let you leave feedback if this is correct.  It looks correct to me solely based on splitting `getAuthenticatedUser` apart into more specific methods